src/components/ActionButton/ToggleActionButton.ts

```diff
@@ -0,0 +1,182 @@
+         * Specifies the button icon when the button is activated.
+         *
+         * Default is the empty string.
+         *
+         * For example, with this SVG markup:
+         *
+         * ```xml
+         * <svg width="1em" height="1em">...</svg>
+         * ```
+         *
+         * The attribute would be set like this:
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-activated-icon='&lt;svg width=&quot;1em&quot; height=&quot;1em&quot;&gt;...&lt;/svg&gt;'></button>
+         * ```
+         */
+        activatedIcon: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the button tooltip when the button is activated.
+         *
+         * Default is the empty string.
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-activated-tooltip='My activated button tooltip'></button>
+         * ```
+         */
+        activatedTooltip: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the button SVG icon.
+         * Note: The SVG markup has to be HTML encoded when set using the HTML attributes.
+         *
+         * Default is the empty string.
+         *
+         * For example, with this SVG markup:
+         *
+         * ```xml
+         * <svg width="1em" height="1em">...</svg>
+         * ```
+         *
+         * The attribute would be set like this:
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-icon='&lt;svg width=&quot;1em&quot; height=&quot;1em&quot;&gt;...&lt;/svg&gt;'></button>
+         * ```
+         */
+        icon: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the button tooltip text.
+         *
+         * Default is the empty string.
+         *
+         * ```html
+         * <button class='CoveoToggleActionButton' data-tooltip='My button tooltip'></button>
+         * ```
+         */
+        tooltip: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the handler called when the button is clicked.
+         *
+         * Default is `null`.
+         *
+         * This option is set in JavaScript when initializing the component.
+         */
+        click: ComponentOptions.buildCustomOption(s => null),
+
+        /**
+         * Specifies the handler called when the button is activated.
+         *
+         * Default is `null`.
+         *
+         * This option is set in JavaScript when initializing the component.
+         */
+        activate: ComponentOptions.buildCustomOption(s => null),
+
+        /**
+         * Specifies the handler called when the button is deactivated.
+         *
+         * Default is `null`.
+         *
+         * This option is set in JavaScript when initializing the component.
+         */
+        deactivate: ComponentOptions.buildCustomOption(s => null)
+    };
+
+    private _isActivated: boolean = false;
+    private innerButton: ActionButton;
```

Comment:
`innerActionButton` I'd say.
Even if that's a bit heavy handed, that leave no room for misundersanding.